src/main/java/org/wildfly/security/cache/CachedIdentity.java

```diff
@@ -40,20 +42,28 @@
     private final transient SecurityIdentity securityIdentity;
 
     /**
-     * Creates a new instance based on the given <code>id</code> and <code>securityIdentity</code>.
+     * Creates a new instance based on the given <code>mechanismName</code> and <code>securityIdentity</code>.
      *
-     * @param id the value that identifies this instance in the cache
      * @param mechanismName the name of the authentication mechanism used to authenticate/authorize the identity
      * @param securityIdentity the identity to cache
      */
     public CachedIdentity(String mechanismName, SecurityIdentity securityIdentity) {
-        checkNotNullParam("mechanismName", mechanismName);
-        checkNotNullParam("securityIdentity", securityIdentity);
-        Principal principal = securityIdentity.getPrincipal();
-        checkNotNullParam("principal", principal);
-        checkNotNullParam("name", principal.getName());
-        this.mechanismName = mechanismName;
-        this.name = principal.getName();
+        this(mechanismName, securityIdentity, () -> checkNotNullParam("securityIdentity", securityIdentity).getPrincipal());
+    }
+
+    /**
+     * Creates a new instance based on the given <code>mechanismName</code> and <code>principal</code>.
+     *
+     * @param mechanismName the name of the authentication mechanism used to authenticate/authorize the identity
+     * @param principal the principal of this cached identity
+     */
+    public CachedIdentity(String mechanismName, Principal principal) {
+        this(mechanismName, null, () -> principal);
```

Comment:
Correct - this lambda is not serializable. However, unless I'm missing something, nothing will ever attempt to serialize it.
This is verified by the test in the associated elytron-web PR.